samples/csharp_dotnetcore/21.corebot-app-insights/AdapterWithErrorHandler.cs

```diff
@@ -6,14 +6,16 @@
 using Microsoft.Bot.Builder.Integration.AspNet.Core;
 using Microsoft.Bot.Builder.TraceExtensions;
 using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.BotBuilderSamples
 {
     public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
     {
-        public AdapterWithErrorHandler(ICredentialProvider credentialProvider, ILogger<BotFrameworkHttpAdapter> logger, IMiddleware middleware, ConversationState conversationState = null)
-            : base(credentialProvider)
+        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger, IMiddleware middleware, ConversationState conversationState = null)
+            : base(configuration, logger)
         {
             Use(middleware);
```

Comment:
I know this is the way it was in the sample, but can we change IMiddleware middleware to TelemetryInitializerMiddleware telemetryInitializer? IMiddleware and the variable name is too broad and can mean anything.

Reply:
Agreed. Modified.